Allow coq-core and coqide-server to be compiled in bytecode-only mode

kit-ty-kate opened the pull request:
e.g. ppc64, x86_32, arm32 or s390x on OCaml 5.0

See https://github.com/ocaml/opam-repository/pull/23584#discussion_r1150586561

ejgallego commented:
@kit-ty-kate this seems broken, likely we'd need to bump the dune lang version?

Alizter commented:
Having a bytecode only version of coqnative seems strange.

SkySkimmer commented:
https://dune.readthedocs.io/en/latest/dune-files.html#linking-modes
`native` is shorthand for `(native exe)`
`best` is not shorthand for anything so can only be used in `(best something)`
so we need to use `exe` (equivalently `(best exe)`)

kit-ty-kate commented:
Fixed. The name of modes doesn't seem to be the same whether or not it is a library or an executable.

I've also added the same fix for coqide-server and now `dune build -p coq-core,coq-stdlib,coqide-server,coq,coqide` now works fine in a bytecode-only switch (tested locally)

ejgallego commented:
@coqbot run full CI

ejgallego commented:
@coqbot: merge now
